src/protocol/thread/pause.ts

```diff
@@ -486,18 +481,28 @@ export class Pause {
 
   async getMouseTarget(x: number, y: number, nodeIds?: string[]) {
     await this.loadMouseTargets();
-    for (const { node, rect, visibility, pointerEvents } of this.mouseTargets!) {
-      const [left, top, right, bottom] = rect;
+    for (let { node, rect, rects, clipBounds, visibility, pointerEvents } of this.mouseTargets!) {
+      if (nodeIds && !nodeIds.includes(node)) {
+        continue;
+      }
+      if (visibility === "hidden" || pointerEvents === "none") {
+        continue;
+      }
       if (
-        (!nodeIds || nodeIds.includes(node)) &&
-        visibility !== "hidden" &&
-        pointerEvents !== "none" &&
-        x >= left &&
-        x <= right &&
-        y >= top &&
-        y <= bottom
+        (clipBounds?.left !== undefined && x < clipBounds.left) ||
+        (clipBounds?.right !== undefined && x > clipBounds.right) ||
+        (clipBounds?.top !== undefined && y < clipBounds.top) ||
+        (clipBounds?.bottom !== undefined && y > clipBounds.bottom)
       ) {
-        return new NodeBoundsFront(this, node, rect);
+        continue;
+      }
+
+      rects ||= [rect];
```

Comment:
I saw the `protocol` mentions that it's either one `rect` or multiple `rects` (backward compatibility). Maybe we can add disjoint unions in TS, something like:
```
{
   rect: Rect
   rects: null
} | {
   rect: null
   rects: Rect[]
}
```
Or add a comment similar to one in `protocol`.

Reply:
`rect` is always defined, only `rects` is optional. I'll add a comment.